direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java

```diff
@@ -710,6 +689,31 @@ private void handleFlushDataBaseOnPosition(
       }
     }
 
+    private StreamElement getStreamElement(Entry<String, Pair<Long, byte[]>> e, int prefixLength) {
+      String keyAttr = e.getKey().substring(prefixLength);
+      final long stamp = e.getValue().getFirst();
+      final String[] parts = keyAttr.split("#");
+      final String key = parts[0];
+      final String attribute = parts[1];
+      final AttributeDescriptor<?> desc = getEntityDescriptor().getAttribute(attribute, true);
+      final StreamElement element =
+          StreamElement.upsert(
+              getEntityDescriptor(),
+              desc,
+              UUID.randomUUID().toString(),
+              key,
+              attribute,
+              stamp,
+              e.getValue().getSecond());
+      return element;
+    }
+
+    private String toConsumedOffset(Partition partition, StreamElement element) {
+      return String.format(
+          "%d-%s#%s:%d",
```

Comment:
NIT: we have already this pattern on at least one more place, maybe we can extract it and avoid situation where somebody change it just on one place.